alerting/src/main/kotlin/org/opensearch/alerting/aggregation/bucketselectorext/BucketSelectorIndices.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.alerting.aggregation.bucketselectorext
+
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.ToXContent.Params
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.search.aggregations.InternalAggregation
+import java.io.IOException
+import java.util.Objects
+
+open class BucketSelectorIndices(
+    name: String?,
+    private var parentBucketPath: String,
+    var bucketIndices: List<Int?>,
+    metaData: Map<String?, Any?>?
+) : InternalAggregation(name, metaData) {
+
+    @Throws(IOException::class)
+    override fun doWriteTo(out: StreamOutput) {
+        out.writeString(parentBucketPath)
+        out.writeIntArray(bucketIndices.stream().mapToInt { i: Int? -> i!! }.toArray())
+    }
+
+    override fun getWriteableName(): String {
+        return name
+    }
+
+    override fun reduce(aggregations: List<InternalAggregation>, reduceContext: ReduceContext): BucketSelectorIndices {
+        throw UnsupportedOperationException("Not supported")
```

Comment:
nit: Should we add the variables in the Exception message? Similar comment for other functions which are not available for the BucketSelectorIndices class.

Reply:
I believe this is consistent with the other classes used for existing pipeline selectors/aggregations and since this is referring to the `reduce()` method itself being unsupported rather than any specific arguments, I think this should be fine.